### PR TITLE
Upgrades to miniz 2.2

### LINKED ISSUE #144: Upgrade to miniz 2.x ?
Hi!
Thanks for a great library!

Do you have any (near future) plans on updating to miniz version 2.x?

Regards,
Mathias

### PR DESCRIPTION
**Pull request by warmwaffles**
This is an attempt at upgrading to miniz 2.2

Closes #144 

Some things changed only slightly. One test fails for me locally right now and that is the `test_permissions.out`

### CODE REVIEW

File: src/zip.h [hunk 1]
```diff
@@ -426,15 +429,6 @@ extern ZIP_EXPORT int zip_extract(const char *zipname, const char *dir,
                                                           void *arg),
                                   void *arg);
 
-/**
- * Sets global CRC-32 checksum function.
- *
- * @param crc32_func crc32 function (init value, buffer, buffer size)
- */
-extern ZIP_EXPORT void
-zip_crc32_func(unsigned long (*crc32_func)(unsigned long crc, const void *buf,
```
Comment (kuba--) [inline src/zip.h:435]: Would be good to have this functionality (for performance reasons).

**Comment by kuba--** [comment]
Check out tests and compilation errors (on windows).

File: src/zip.h (hunk 1 above)
Comment (kuba--) [reply src/zip.h:435]: ...someone asked for this functionality so I've added. It's  never been part of miniz.

Comment (warmwaffles) [reply src/zip.h:435]: Did you patch it into miniz directly?

Comment (kuba--) [reply src/zip.h:435]: yes, our miniz, contains many _patches_/additions

**Comment by kuba--** [comment]
Regarding permissions, we need to keep the same permissions mode (that's why `stat` function was used). A file permissions like other attributes are saved in zip dictionary.

**Comment by warmwaffles** [comment]
> Regarding permissions, we need to keep the same permissions mode (that's why stat function was used). A file permissions like other attributes are saved in zip dictionary.

In this PR I didn't modify anything around the stat functionality. Because the miniz 1.15 is inlined as a header only, this wrapper is using a "private" function in the official 2.2 release. Thus, just including it as an amalgamated single header file, we've regained access to the stat function.

I'm going to go ahead and close this, feel free to pick any changes if you want.

**Pull request closed without merging**